src/compiler/optimize.jl

```diff
@@ -22,273 +22,151 @@ end
-        basic_alias_analysis!(pm)
-        cpu_features_tm!(pm, tm)
-
-        LLVM.API.LLVMAddGlobalOptimizerPass(pm) # Extra
-        gvn!(pm) # Extra
-        LLVM.run!(pm, mod)
+        run!(pb, mod, tm)
     end
 
-    rewrite_generic_memory!(mod)
-    
-    ModulePassManager() do pm
-        add_library_info!(pm, triple(mod))
-        add_transform_info!(pm, tm)
-
-        scoped_no_alias_aa!(pm)
-        type_based_alias_analysis!(pm)
-        basic_alias_analysis!(pm)
-        cpu_features_tm!(pm, tm)
-
-        instruction_combining!(pm)
-        jl_inst_simplify!(pm)
-        cfgsimplification!(pm)
-        scalar_repl_aggregates_ssa!(pm) # SSA variant?
-        instruction_combining!(pm)
-        jl_inst_simplify!(pm)
-        jump_threading!(pm)
-        correlated_value_propagation!(pm)
-        instruction_combining!(pm)
-        jl_inst_simplify!(pm)
-        reassociate!(pm)
-        early_cse!(pm)
-        alloc_opt_tm!(pm, tm)
-        loop_idiom!(pm)
-        loop_rotate!(pm)
-
-        loop_optimizations_tm!(pm, tm)
-
-        instruction_combining!(pm)
-        jl_inst_simplify!(pm)
-        ind_var_simplify!(pm)
-        loop_deletion!(pm)
-        loop_unroll!(pm)
-        alloc_opt_tm!(pm, tm)
-        scalar_repl_aggregates_ssa!(pm) # SSA variant?
-        gvn!(pm)
-
-        # This InstCombine needs to be after GVN
-        # Otherwise it will generate load chains in GPU code...
-        instruction_combining!(pm)
-        jl_inst_simplify!(pm)
-        mem_cpy_opt!(pm)
-        sccp!(pm)
-        instruction_combining!(pm)
-        jl_inst_simplify!(pm)
-        jump_threading!(pm)
-        dead_store_elimination!(pm)
-        alloc_opt_tm!(pm, tm)
-        cfgsimplification!(pm)
-        loop_idiom!(pm)
-        loop_deletion!(pm)
-        jump_threading!(pm)
-        correlated_value_propagation!(pm)
-        # SLP_Vectorizer -- not for Enzyme
-
-        LLVM.run!(pm, mod)
-
-        aggressive_dce!(pm)
-        instruction_combining!(pm)
-        jl_inst_simplify!(pm)
-        # Loop Vectorize -- not for Enzyme
-        # InstCombine
-
-        # GC passes
-        barrier_noop!(pm)
-        gc_invariant_verifier_tm!(pm, tm, false)
-
-        # FIXME: Currently crashes printing
-        cfgsimplification!(pm)
-        instruction_combining!(pm) # Extra for Enzyme
-        jl_inst_simplify!(pm)
-        LLVM.run!(pm, mod)
-    end
-    
     # Globalopt is separated as it can delete functions, which invalidates the Julia hardcoded pointers to
     # known functions
-    ModulePassManager() do pm
-        add_library_info!(pm, triple(mod))
-        add_transform_info!(pm, tm)
-
-        scoped_no_alias_aa!(pm)
-        type_based_alias_analysis!(pm)
-        basic_alias_analysis!(pm)
-        cpu_features_tm!(pm, tm)
-
-        LLVM.API.LLVMAddGlobalOptimizerPass(pm) # Exxtra
-        gvn!(pm) # Exxtra
-        LLVM.run!(pm, mod)
+    @dispose pb = NewPMPassBuilder() begin
+        add!(pb, NewPMAAManager()) do aam
```

Comment:
lets see if we can just do this one, for example, then the prev module pipeline, etc, til we figure out the failure